### ISSUE TITLE
Invoke-PortalUpgradeScript -cf XXX should be Invoke-PortalUpgradeScript -PortalConfig XXX

### ISSUE THREAD

**Issue by edgjeromelaurent** [Esri/arcgis-powershell-dsc#393]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.3.2

### Affected Resource(s)

* ArcGIS_Portal

### Configuration Files

```
{
    "AllNodes": [
        {
            "NodeName": "arcgis_portal_servername.arcgis_servername_suffixe",
            "Role": [
                "Portal",
                "PortalWebAdaptor"
            ],
            "SslCertificates": [
                {
                    "Path": "D:\\_sources\\arcgis_num_qualiparc\\arcgis_install_conf\\certificats\\arcgis_portal_servername.pfx",
                    "Password": "arcgis_certificate_password",
                    "CNameFQDN": "arcgis_portal_servername.arcgis_servername_suffixe",
                    "Target": [
                        "Portal"
                    ]
                }
            ]
        }
    ],
    "ConfigData": {
        "Version": "10.9.1",
        "OldVersion": "10.8.1",
        "PortalContext": "arcgis_portal_context",
        "ServerRole": "GeneralPurposeServer",
        "Credentials": {
            "ServiceAccount": {
                "UserName": "arcgis_runas_username",
                "Password": "arcgis_runas_password",
                "IsDomainAccount": true
            }
        },
        "Portal": {
            "LicenseFilePath": "D:\\_sources\\arcgis_num_qualiparc\\arcgis_install_conf\\licences\\portal_for_arcgis\\arcgis_portal_licencefilename_TRANSLATE_VERSION_NEW",
            "PortalLicenseUserTypeId": "creatorUT",
            "Installer": {
                "Path": "D:\\_sources\\arcgis_num_qualiparc\\arcgis_versions\\TRANSLATE_VERSION_NEW\\portal_for_arcgis-TRANSLATE_VERSION_NEW.exe",
                "InstallDir": "D:\\programs\\ArcGIS\\Portal",
                "ContentDir": "D:\\data\\arcgis_num_qualiparc\\arcgisportal",
                "PatchesDir": "D:\\_sources\\arcgis_num_qualiparc\\arcgis_versions\\TRANSLATE_VERSION_NEW\\patchs\\portal_for_arcgis"
            },
            "ContentDirectoryLocation":"D:\\data\\arcgis_num_qualiparc\\arcgisportal\\content",
            "EnableAutomaticAccountCreation": "False",
            "DefaultRoleForUser": "account_user",
            "DefaultUserLicenseTypeIdForUser": "creatorUT",
            "ExternalLoadBalancer": "arcgis_externalloadbalancer",
            "PortalAdministrator": {
                "UserName": "arcgis_portal_username",
                "Email": "tickets.geomatique@etat.ge.ch",
                "Password": "arcgis_portal_password",
                "SecurityQuestionIndex": 1,
                "SecurityAnswer": "vanilla"
            },
            "EmailSettings":{
                "SMTPServerAddress":"smtp.etat-ge.ch",
                "From":"tickets.geomatique@etat.ge.ch",
                "Label":"SMTP EDG",
                "AuthenticationRequired":0,
                "EncryptionMethod":"NONE",
                "SMTPPort":"25",
                "UserName": "",
                "Password": ""
            }
        },
        "WebAdaptor": {
            "AdminAccessEnabled": true,
            "Installer": {
                "Path": "D:\\_sources\\arcgis_num_qualiparc\\arcgis_versions\\TRANSLATE_VERSION_NEW\\webadaptor-TRANSLATE_VERSION_NEW.exe"
            },
            "WebSiteId": 1
        }
    }
}
```

### Expected Behavior

Portal should be upgraded.

### Actual Behavior

Bad parameter name stop upgrade.

### Steps to Reproduce

Run 'Invoke-ArcGISConfiguration' with the following command:

`Invoke-ArcGISConfiguration -ConfigurationParametersFile D:\path\config_portal.json -Mode Upgrade -MappedDriveOverrideFlag $False
`

Error :
```
STDERR:

Invoke-PortalUpgradeScript : A parameter cannot be found that matches parameter name 'cf'.
At C:\Program Files\WindowsPowerShell\Modules\ArcGIS\ArcGIS.psm1:1781 char:55
+                 $JobFlag = Invoke-PortalUpgradeScript -cf $NonEnterpi ...
+                                                       ~~~
    + CategoryInfo          : InvalidArgument: (:) [Invoke-PortalUpgradeScript], ParameterBinding 
   Exception
    + FullyQualifiedErrorId : NamedParameterNotFound,Invoke-PortalUpgradeScript

```

### References

https://github.com/Esri/arcgis-powershell-dsc/blob/6ee6c84c096dcb30b9acf80a022bddf704873594/Modules/ArcGIS/ArcGIS.psm1#L1781

cf parameter should be PortalConfig

**Comment by cameronkroeker** [comment]
Hello @edgjeromelaurent,

Great find, and thanks for sharing the workaround! We will get this fixed in a future release of the module.

**Workaround:** In line 1781 of ArcGIS.psm1 change`-cf` to `-PortalConfig`
Update:
https://github.com/Esri/arcgis-powershell-dsc/blob/6ee6c84c096dcb30b9acf80a022bddf704873594/Modules/ArcGIS/ArcGIS.psm1#L1781

To:
```powershell
$JobFlag = Invoke-PortalUpgradeScript -PortalConfig $NonEnterpiseConfig -Credential $Credential -UseWinRMSSL $UseWinRMSSL -DebugMode $DebugMode -EnableMSILogging $EnableMSILoggingMode
```

Thanks,
Cameron K.

**Comment by cameronkroeker** [comment]
Hi @edgjeromelaurent,

This issue has been addressed in the latest release, v4.0.0:

https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v4.0.0

Thanks,
Cameron K.
